devsearch/preprocessing/extract.py

```diff
@@ -0,0 +1,21 @@
+from dulwich import porcelain
+from dulwich.repo import Repo
+
+
+def handle_commit(commit):
+    print(f'author: {commit.author.decode()}, sha: {commit.id.decode()}')
+
+
+def extract_repo(git_path: str, target_path: str, should_clone=False, limit=10):
+    try:
+        if should_clone:
+            porcelain.clone(git_path, target_path)
+    except Exception as e:
```

Comment:
very general exception. Let's narrow it down to exception that can occur because of:
* repo exist
* bad URL
* etc